sound/soc/sof/topology.c

```diff
@@ -1382,6 +1382,14 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 
 	w->dobj.private = swidget;
 	list_add(&swidget->list, &sdev->widget_list);
+
+	/* load 3rd party module library if needed */
+	if (sdev->ipc->ops->fw_loader->load_library) {
+		ret = sdev->ipc->ops->fw_loader->load_library(sdev, swidget);
```

Comment:
I am officially confused about the two calls to load_library, here and there https://github.com/thesofproject/linux/pull/3720/commits/8f62cf663e27df09feb62a9778e1cd6d4b1f2c6f#diff-9cbcb1a6c800a373c6da46cdcf07ef799b13e2634d64e9aafb74f4a83302e773R323

Reply:
oh they are different. One is the top-level topology widget op to load the library needed for it and the other is the platform-specific DSP op to actually load the library in the FW